dll/shellext/fontext/CFontExt.cpp

```diff
@@ -533,71 +527,23 @@ STDMETHODIMP CFontExt::Drop(IDataObject* pDataObj, DWORD grfKeyState, POINTL pt,
         return E_FAIL;
     }
 
-    BOOL bOK = TRUE;
-    CAtlArray<CStringW> FontPaths;
+    CAtlArray<PCUIDLIST_RELATIVE> apidl;
     for (UINT n = 0; n < cida->cidl; ++n)
     {
         PCUIDLIST_RELATIVE pidlRelative = HIDA_GetPIDLItem(cida, n);
         if (!pidlRelative)
-            continue;
-
-        PIDLIST_ABSOLUTE pidl = ILCombine(pidlParent, pidlRelative);
-        if (!pidl)
-        {
-            ERR("ILCombine failed\n");
-            bOK = FALSE;
-            break;
-        }
-
-        WCHAR szPath[MAX_PATH];
-        BOOL ret = SHGetPathFromIDListW(pidl, szPath);
-        ILFree(pidl);
-
-        if (!ret)
-        {
-            ERR("SHGetPathFromIDListW failed\n");
-            bOK = FALSE;
-            break;
-        }
-
-        if (PathIsDirectoryW(szPath))
-        {
-            ERR("PathIsDirectory\n");
-            bOK = FALSE;
-            break;
-        }
-
-        LPCWSTR pchDotExt = PathFindExtensionW(szPath);
-        if (!IsFontDotExt(pchDotExt))
-        {
-            ERR("'%S' is not supported\n", pchDotExt);
-            bOK = FALSE;
-            break;
-        }
+            return E_FAIL;
 
-        FontPaths.Add(szPath);
+        apidl.Add(pidlRelative);
     }
 
-    if (!bOK)
-        return E_FAIL;
-
-    CRegKey keyFonts;
-    if (keyFonts.Open(FONT_HIVE, FONT_KEY, KEY_WRITE) != ERROR_SUCCESS)
+    CStringW strMessage;
+    if (InstallFontFiles(strMessage, pidlParent, cida->cidl, &apidl[0]) != S_OK)
```

Comment:
This will explode if `cidl` == `0` (because `apidl` is empty)

Reply:
Added a check. Thanks.